### PR TITLE
fix #317, allow user to customize the akka system config for appmaster and executor

### LINKED ISSUE #317: Allow AppMaster to configure customized akka config

### PR DESCRIPTION
**Pull request by clockfly**
[Review on Reviewable](https://reviewable.io/reviews/intel-hadoop/gearpump/318)

### CODE REVIEW

File: core/src/main/scala/org/apache/gearpump/cluster/master/AppMasterLauncher.scala [hunk 1]
```diff
@@ -38,15 +38,19 @@ import org.slf4j.Logger
...
 
+  import app._
+
   val scheduler = context.system.scheduler
   val systemConfig = context.system.settings.config
   val TIMEOUT = Duration(15, TimeUnit.SECONDS)
-  val appMaster = app.appMaster
-  val userConfig = app.conf
+
+  val appMasterAkkaConfig: Config = if (appMasterClusterConfig == null) ConfigFactory.empty() else appMasterClusterConfig.getConfig
```
Comment (manuzhang) [inline core/src/main/scala/org/apache/gearpump/cluster/master/AppMasterLauncher.scala]: maybe cleaner with "Option(...).getOrElse(...)"

File: core/src/main/scala/org/apache/gearpump/cluster/Application.scala [hunk 2]
```diff
@@ -27,15 +27,16 @@ import org.apache.gearpump.jarstore.JarFileContainer
...
  * @param name: The name of this application
- * @param conf: user configuration.
+ * @param userConfig: user configuration.
  * @param appMaster: The class name of AppMaster Actor
+ * @param appMasterClusterConfig: The akka config source used to create the appmaster actor system
  *    The AppMaster must have a constructor this(appContext : AppMasterContext, app : Application)
  */
 
-final class Application(val name : String, val appMaster : String, val conf: UserConfig) extends Serializable
+final class Application(val name : String, val appMaster : String, val userConfig: UserConfig, val appMasterClusterConfig: ClusterConfigSource = null) extends Serializable
```
Comment (huafengw) [inline core/src/main/scala/org/apache/gearpump/cluster/Application.scala]: What is appMasterClusterConfig used for and what's the difference between userConfig here? I think maybe it's kind of difficult for a user to distinguish these kinds of config.

File: core/src/main/scala/org/apache/gearpump/cluster/master/AppMasterLauncher.scala (hunk 1 above)
Comment (clockfly) [inline core/src/main/scala/org/apache/gearpump/cluster/master/AppMasterLauncher.scala]: OK

File: core/src/main/scala/org/apache/gearpump/cluster/Application.scala (hunk 2 above)
Comment (clockfly) [inline core/src/main/scala/org/apache/gearpump/cluster/Application.scala]: appMasterClusterConfig is used to initialize the akka system used by appmaster.

Will add more comments.

**Comment by huafengw** [comment]
It seems that the WorkerSpec could not pass in this PR, still figuring it out.

**Comment by clockfly** [comment]
fixed.

It is due to future {process.exitValue} will occupy the dispatcher thread pool, so input message of actors will not be dispatched.

**Comment by huafengw** [comment]
+1

**Pull request merged**